java/ql/src/semmle/code/java/dataflow/internal/TaintTrackingUtil.qll

```diff
@@ -93,6 +98,81 @@ private module Cached {
 
 import Cached
 
+private module StoreTaintSteps {
+  private import semmle.code.java.dataflow.TaintTracking
+  private import semmle.code.java.dataflow.TaintTracking2
+
+  private class StoreTaintConfig extends TaintTracking::Configuration {
+    StoreTaintConfig() { this instanceof TaintTracking::Configuration or none() }
```

Comment:
This is somewhat unusual, could you explain what's going on here? How does this not cause collisions with other implementations of `TaintTracking::Configuration` actually used in the query?

Reply:
This is indeed unusual 😉 And yes, it collides with every other configuration. It's a hack to globally modify all other configurations with code that is itself configuration-dependent.

Reply:
Why use this vs implementing `AdditionalTaintStep`? Isn't the API for `isAdditionalTaintStep` exactly the same and doesn't creating implementations of `AdditionalTaintStep` have exactly the same effect?

Reply:
No. The `Configuration::isAdditionalTaintStep` predicate is configuration-specific, whereas `AdditionalTaintStep` is configuration-independent (i.e. global and applies to all configs).

Reply:
There are only two copies `TaintTracking` at the moment, so this should be all of them.

Reply:
Besides, this is a temporary measure, since it is quite costly in terms of performance.